metadata-ingestion/src/datahub/configuration/config_loader.py

```diff
@@ -10,7 +11,7 @@
 
 
 def resolve_element(element: str) -> str:
-    if element.startswith("${") & element.endswith("}"):
+    if re.search("(\$\{).+(\})", element):  # noqa: W605
         return expandvars(element, nounset=True)
     elif element.startswith("$"):
```

Comment:
Do we still need this conditional?

Reply:
Yes. Otherwise, when there is $ in value, lets say in password and no need of expansion, we dont want to execute expandvars.

Reply:
But what if $ is the first letter of a password?

Eg. 

password: $something 

Will this variable be mistakenly expanded?

Reply:
if there is an env_var match for these cases, then it will be expanded, otherwise the original string will be retained. Which allows us to support existing folks that are using `$VAR` form for existing variables. Added a test to validate